bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -390,21 +391,50 @@ impl Script {
     /// broadcastable on today's Bitcoin network.
+    ///
+    /// Dust depends on the -dustrelayfee value of the Bitcoin Core node you are broadcasting to.
+    /// This function uses the default value of 0.00003 BTC/kB (3 sat/vByte).
+    ///
+    /// To use a custom value, use [`dust_value_custom`].
+    ///
+    /// [`dust_value_custom`]: Script::dust_value_custom
     pub fn dust_value(&self) -> crate::Amount {
+        self.dust_value_inner(DUST_RELAY_TX_FEE as u64)
+    }
+
+    /// Returns the minimum value an output with this script should have in order to be
+    /// broadcastable on today's Bitcoin network.
+    ///
+    /// Dust depends on the -dustrelayfee value of the Bitcoin Core node you are broadcasting to.
+    /// This function lets you set the fee rate used in dust calculation.
+    ///
+    /// The current default value in Bitcoin Core (as of v26) is 3 sat/vByte.
+    ///
+    /// To use the default Bitcoin Core value, use [`dust_value`].
+    ///
+    /// [`dust_value`]: Script::dust_value
+    pub fn dust_value_custom(&self, dust_relay_fee: FeeRate) -> crate::Amount {
+        self.dust_value_inner(dust_relay_fee.to_sat_per_kwu() * 4)
+    }
+
+    fn dust_value_inner(&self, dust_relay_fee: u64) -> crate::Amount {
         // This must never be lower than Bitcoin Core's GetDustThreshold() (as of v0.21) as it may
         // otherwise allow users to create transactions which likely can never be broadcast/confirmed.
-        let sats = DUST_RELAY_TX_FEE as u64 / 1000 * // The default dust relay fee is 3000 satoshi/kB (i.e. 3 sat/vByte)
-        if self.is_op_return() {
-            0
-        } else if self.is_witness_program() {
-            32 + 4 + 1 + (107 / 4) + 4 + // The spend cost copied from Core
-            8 + // The serialized size of the TxOut's amount field
-            self.consensus_encode(&mut sink()).expect("sinks don't error") as u64 // The serialized size of this script_pubkey
-        } else {
-            32 + 4 + 1 + 107 + 4 + // The spend cost copied from Core
-            8 + // The serialized size of the TxOut's amount field
-            self.consensus_encode(&mut sink()).expect("sinks don't error") as u64 // The serialized size of this script_pubkey
-        };
+        let sats = dust_relay_fee *
+            if self.is_op_return() {
+                0
+            } else if self.is_witness_program() {
+                32 + 4 + 1 + (107 / 4) + 4 + // The spend cost copied from Core
+                8 + // The serialized size of the TxOut's amount field
+                self.consensus_encode(&mut sink()).expect("sinks don't error") as u64 // The serialized size of this script_pubkey
+            } else {
+                32 + 4 + 1 + 107 + 4 + // The spend cost copied from Core
+                8 + // The serialized size of the TxOut's amount field
+                self.consensus_encode(&mut sink()).expect("sinks don't error") as u64 // The serialized size of this script_pubkey
```

Comment:
This doesn't perform checked add but I think overflow is impossible because IIRC the maximum length of a slice in rust is actually `(1 << (size_of::<usize>() * 8 - 1)) - 1` and all the constants we add are way less than that even for 16 bit platforms.